src/gfx/render_text.hpp

```diff
@@ -63,6 +64,6 @@ void clear_scale_aware_text(sf::RenderTexture& texture);
 void win_draw_string(sf::RenderTarget& dest_window,rectangle dest_rect,std::string str,eTextMode mode,TextStyle style,bool right_align = false);
 void win_draw_string(sf::RenderTarget& dest_window,rectangle dest_rect,std::string str,eTextMode mode,TextStyle style, break_info_t break_info,bool right_align = false);
 break_info_t calculate_line_wrapping(rectangle dest_rect, std::string str, TextStyle style);
-size_t string_length(std::string str, TextStyle style, short* height = nullptr);
+size_t string_length(std::string str, TextStyle& style, short* height = nullptr);
```

Comment:
The style should be a const reference. The measurement cache can be declared mutable.